crates/libs/windows/src/core/agile_reference.rs

```diff
@@ -0,0 +1,32 @@
+use super::*;
+use bindings::*;
+use core::marker::PhantomData;
+
+pub struct AgileReference<T> {
+    reference: IAgileReference,
+    _marker: PhantomData<T>,
+}
+
+impl<T: Interface> AgileReference<T> {
+    pub fn new<'a>(from_ref: &'a T) -> Result<Self>
+    where
+        &'a T: IntoParam<'a, IUnknown>,
+    {
+        let reference = unsafe { RoGetAgileReference(AGILEREFERENCE_DEFAULT, &T::IID, from_ref)? };
+        Ok(Self { reference, _marker: Default::default() })
+    }
+
+    pub fn resolve(&self) -> Result<T> {
```

Comment:
This should simply be a call to `Resolve` but is currently blocked by this issue:

https://github.com/microsoft/win32metadata/issues/787